apps/tests/Robot/src/robottest/RobotBuilder.java

```diff
@@ -350,24 +343,21 @@ private boolean checkColor(int value, Color expected) {
         return false;
     }
 
-    public Popup robotScreenTest(final TextField result, Stage stage){
-
+    public Popup robotScreenTest(final TextField result, Stage stage) {
         Bounds bounds = rec1.localToScreen(new BoundingBox(0, 0,
                 rec1.getBoundsInParent().getWidth(),
                 rec1.getBoundsInParent().getHeight()));
 
         int x = 50 + (int) bounds.getMinX();
         int y = 50 + (int) bounds.getMinY();
-        int[] intArr = null;
         boolean correct = true;
-        Robot robot = com.sun.glass.ui.Application.GetApplication().createRobot();
+        final Robot robot = new Robot();
         int width = 160;
         int height = 160;
-        final Buffer buff = robot.getScreenCapture(x, y, width, height).getPixels();
-        if ((buff instanceof IntBuffer)&&(buff.hasArray())) {
-            intArr =((IntBuffer) buff).array();
-        }
-
+        int[] intArr = new int[width * height * 4];
+        PixelReader pixelReader = robot.getScreenCapture(null, x, y, width, height).getPixelReader();
+        WritablePixelFormat<IntBuffer> format = WritablePixelFormat.getIntArgbInstance();
+        pixelReader.getPixels(0, 0, width, height, format, intArr, 0, width * 4);
```

Comment:
The `* 4` is wrong, both here and in the array allocation. The scanline stride should be `width` and you only need room for `width * height` pixels.

Reply:
Fixed.